src/SFA.DAS.Employer.PR.Web/Models/Session/AddTrainingProvidersSessionModel.cs

```diff
@@ -4,4 +4,7 @@ public class AddTrainingProvidersSessionModel
 {
     public long? LegalEntityId { get; set; }
     public string? LegalName { get; set; }
-}
+    public int NumberOfLegalEntities { get; set; }
```

Comment:
This will become redundant once you move the logic correctly in SelectLegalEntityController

Reply:
No it is still needed for logic in the navigation for the back button.  The issue is when you have 1 and 1 only legal entity, and are on the select training provider page and hit 'back', and it redirects back to the 'SelectLegalEntity' page... which immediately redirects back to Select training provider page. By setting this early on, you can use it to set the correct back link on the select training provider page. I hope that makes sense.

Reply:
Yes that makes sense.